views/default/forms/usersettings/save.php

```diff
@@ -5,10 +5,18 @@
  * Plugins should extend "forms/account/settings" to add to the settings.
  */
 
-$form_body = elgg_view("forms/account/settings", $vars);
+$user = elgg_get_page_owner_entity();
 
-$form_body .= '<div class="elgg-foot">';
-$form_body .= elgg_view('input/submit', array('value' => elgg_echo('save')));
-$form_body .= '</div>';
+$form_body = elgg_view('forms/account/settings', $vars);
+
+$submit = elgg_view('input/submit', ['value' => elgg_echo('save')]);
+$hidden = '';
```

Comment:
> IMO views should only take care of displaying the data. It's the responsibility of the calling script to make sure that the correct data is provided for the view.

That's a bold assumption, that i disagree with. A lot of elgg bugs in the past were related to views assuming everything is available for use. That is not always the case, see example here #6415